Contrib: update opus to 1.6.1

hnsteyding opened the pull request:
**Opus 1.6:**

Opus 1.6 builds on the new ML-based features introduced in Opus 1.5. Major changes since 1.5 include:
- A new wideband-to-fullband bandwidth extension (BWE) module
- Support for 96 kHz audio with Opus HD
- Significant improvement to Deep Redundancy (DRED)
- A new 24-bit encoder/decoder API

**Tested on:**
- [X] Windows 10+  (via MinGW)
- [ ] macOS 10.13+
- [X] Ubuntu Linux

hnsteyding committed:
Contrib: update opus to 1.6

**Opus 1.6:**

Opus 1.6 builds on the new ML-based features introduced in Opus 1.5.
Major changes since 1.5 include:
- A new wideband-to-fullband bandwidth extension (BWE) module
- Support for 96 kHz audio with Opus HD
- Significant improvement to Deep Redundancy (DRED)
- A new 24-bit encoder/decoder API

**Tested on:**
- [X] Windows 10+  (via MinGW)
- [ ] macOS 10.13+
- [X] Ubuntu Linux

hnsteyding commented:
Some notes:

- Differences between source links:
  https://downloads.xiph.org/releases/opus/opus-1.6.tar.gz (34,6MB - tar ball contains additional folder "dnn") 
  https://gitlab.xiph.org/xiph/opus/-/archive/v1.6/opus-v1.6.tar.gz (4,7MB)

- Support for 96 kHz audio with Opus HD:
  Log entry:
`  [10:35:06] work: sanitizing track 1 unsupported samplerate 96000 Hz to 48 kHz`
[Vox-Titel-1-Winkel-1-16.12.2025-10-34-Uhr-VP9-VP9-NVENC Opus 96 MP4_encode_12.16.2025 10-35-05.txt](https://github.com/user-attachments/files/24192061/Vox-Titel-1-Winkel-1-16.12.2025-10-34-Uhr-VP9-VP9-NVENC.Opus.96.MP4_encode_12.16.2025.10-35-05.txt)
  Are there some modifikation necessary for HandBrake ?

galad87 commented:
Yes probably the new high res feature will need a few tweaks in libavcodec.

hnsteyding commented:
Okay. 
Unfortunately, I lack the knowledge to make any adjustments here.

bradleysepos commented:
See also #659.

hnsteyding committed:
activate OpusHD support

hnsteyding committed:
add space between compile options

hnsteyding committed:
Add files via upload

hnsteyding commented:
Make some further changes and patching ffmpeg to set maximum saple rate to 96000Hz (96kHz).

HB allegedly use sample rate 96000Hz (96kHz):
[Vox-Titel-1-Winkel-1-19.12.2025-14-29-Uhr-nvenc_h264-H.264-NVENC Opus 96 MP4_encode_12.19.2025 14-29-50.txt](https://github.com/user-attachments/files/24258778/Vox-Titel-1-Winkel-1-19.12.2025-14-29-Uhr-nvenc_h264-H.264-NVENC.Opus.96.MP4_encode_12.19.2025.14-29-50.txt)

But output video file has still sample rate of 48000Hz (48kHz):
[Vox-Titel-1-Winkel-1-19.12.2025-14-29-Uhr-nvenc_h264-H.264-NVENC Opus 96 MP4.mp4.txt](https://github.com/user-attachments/files/24258804/Vox-Titel-1-Winkel-1-19.12.2025-14-29-Uhr-nvenc_h264-H.264-NVENC.Opus.96.MP4.mp4.txt)

Maybe someone have a look at this ?!

hnsteyding committed:
Rename A23-OpusHD.patch file to A23-avcodec-set-samplerate-to-96khz-for-OpusHD.patch

hnsteyding committed:
Add libopusenc 0.3 - enable dred and osce support

hnsteyding commented:
Seems not easy to activate/use OpusHD:
https://hydrogenaudio.org/index.php?PHPSESSID=acf0pu3il82qv31fcq8d43k17m&topic=128880.msg1074374#msg1074374

https://opus-codec.org/demo/opus-1.6/:

> To enable Opus HD support, use the `--enable-qext` configure option when building Opus.
> To encode using Opus HD, use the -qext option in opus_demo, or use _OPUS_SET_QEXT(1)_ with the encoder API.
> If built with Opus HD support, the decoder will automatically use any Opus HD layer it finds, unless _OPUS_SET_IGNORE_EXTENSIONS(1)_ is used.
> ...
> The header files were previously using internal macros starting with an __opus prefix,
> which violates the C specification saying that identifiers beginning with a double-underscore are reserved.
> This has now been fixed. Unfortunately, libopusenc was (wrongly) relying on those internal macros,
> so anyone compiling with libopusenc (binaries are unaffected) will need to update to libopusenc 0.3.

hnsteyding commented:
It seems that sample rate are generally downsampled to 48000Hz (48kHz) if MP4 format is used no matter which video encoder to choose.

If WebM or MKV fomat and video encoder VP8, VP9 or AV1 is used output file has a sample rate of 96000Hz (96kHz):
[Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM - Opus 96kHz Samplerate_encode_12.24.2025 02-12-49.txt](https://github.com/user-attachments/files/24328723/Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM.-.Opus.96kHz.Samplerate_encode_12.24.2025.02-12-49.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-11-Uhr-VP8-VP8-WebM - Opus 96kHz Samplerate_encode_12.24.2025 02-11-44.txt](https://github.com/user-attachments/files/24328724/Vox-Titel-1-Winkel-1-24.12.2025-02-11-Uhr-VP8-VP8-WebM.-.Opus.96kHz.Samplerate_encode_12.24.2025.02-11-44.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-09-Uhr-svt_av1-AV1-WebM - Opus 96kHz Samplerate_encode_12.24.2025 02-10-10.txt](https://github.com/user-attachments/files/24328725/Vox-Titel-1-Winkel-1-24.12.2025-02-09-Uhr-svt_av1-AV1-WebM.-.Opus.96kHz.Samplerate_encode_12.24.2025.02-10-10.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-11-Uhr-VP8-VP8-WebM - Opus 96kHz Samplerate.webm.txt](https://github.com/user-attachments/files/24328726/Vox-Titel-1-Winkel-1-24.12.2025-02-11-Uhr-VP8-VP8-WebM.-.Opus.96kHz.Samplerate.webm.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-09-Uhr-svt_av1-AV1-WebM - Opus 96kHz Samplerate.webm.txt](https://github.com/user-attachments/files/24328727/Vox-Titel-1-Winkel-1-24.12.2025-02-09-Uhr-svt_av1-AV1-WebM.-.Opus.96kHz.Samplerate.webm.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM - Opus 96kHz Samplerate.mp4.txt](https://github.com/user-attachments/files/24328728/Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM.-.Opus.96kHz.Samplerate.mp4.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM - Opus 96kHz Samplerate.mkv.txt](https://github.com/user-attachments/files/24328729/Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM.-.Opus.96kHz.Samplerate.mkv.txt)
[Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM - Opus 96kHz Samplerate_encode_12.24.2025 02-16-11.txt](https://github.com/user-attachments/files/24328730/Vox-Titel-1-Winkel-1-24.12.2025-02-12-Uhr-VP9-VP9-WebM.-.Opus.96kHz.Samplerate_encode_12.24.2025.02-16-11.txt)

hnsteyding commented:
BTW:
- Compiling libopusenc 0.3 not needed
- Using small tar ball from https://gitlab.xiph.org/xiph/opus/-/archive/v1.6/opus-v1.6.tar.gz (4,7MB) is sufficient.
- Enable DRED and OSCE support during compilation of Opus 1.6 not needed to enable sample rate of 96000Hz (96kHz) - only add `--enable-qext` configure option when building Opus.

This reduce tar ball download size and also `hb.dll` and` HandBrakeCLI` size.

hnsteyding committed:
Remove libopusenc - not needed

hnsteyding committed:
Remove libopusenc from HandBrake

BTW:
- Compiling libopusenc 0.3 not needed
- Using small tar ball from https://gitlab.xiph.org/xiph/opus/-/archive/v1.6/opus-v1.6.tar.gz (4,7MB) is sufficient.
- Enable DRED and OSCE support during compilation of Opus 1.6 not needed to enable sample rate of 96000Hz (96kHz) - only add `--enable-qext` configure option when building Opus.

This reduce tar ball download size and also `hb.dll` and` HandBrakeCLI` size.

hnsteyding committed:
Merge branch 'HandBrake:master' into Contrib-update-opus-to-1.6

hnsteyding commented:
Output file has also a sample rate of 96000Hz (96kHz) if choose MKV fomat and video encoder FFV1, H.264, H.265 or MPEG-4.

hnsteyding commented:
Strange.
I use MediaInfo V25.04 on previous creeated MediaInfo log files which shows 48kHz if MP4 format is selected.

After I've updated MediaInfo to V25.10 now MP4 output files shows correct sample rate of 96kHz:
[Vox-Titel-1-Winkel-1-27.12.2025-23-59-Uhr-VP9-VP9-WebM - Opus 96kHz Samplerate.mp4.txt](https://github.com/user-attachments/files/24357256/Vox-Titel-1-Winkel-1-27.12.2025-23-59-Uhr-VP9-VP9-WebM.-.Opus.96kHz.Samplerate.mp4.txt)

hnsteyding committed:
Remove LIBOPUSENC from ffmpeg deps

galad87 commented:
Can you split the lib update to a different PR? So we can merge it until the libavcodec part is ready.

hnsteyding commented:
Sorry, i don't understand.
Did you mean:
One PR only for update contrib and one PR only for patching FFMPEG and `common.c` ?

galad87 commented:
Yes.

hnsteyding committed:
Merge branch 'HandBrake:master' into Contrib-update-opus-to-1.6

hnsteyding committed:
Delete file A23-avcodec-set-samplerate-to-96khz-for-OpusHD.patch

hnsteyding committed:
Roll back 96kHz support  for ffmpeg - see PR #7561

hnsteyding committed:
Merge branch 'HandBrake:master' into Contrib-update-opus-to-1.6

hnsteyding committed:
Update opus to 1.6.1

Opus 1.6.1 fixes several minor issues that were discovered since the 1.6 release.
